src/esn.calendar.libs/app/services/event-service.js

```diff
@@ -575,4 +581,60 @@ function calEventService(
       return new CalendarShell(ICAL.Component.fromString(response.data));
     });
   }
+
+  function duplicateEvent(event) {
```

Comment:
duplicateEvent may be in EventUtils as EventService is more dedicated to API call handling.

WDYT ?

Reply:
Yes I agree.

Reply:
I have found a better place to put the function than `calEventUtils`, which is `calEventDuplicateService`.